library/aes.c

```diff
@@ -1110,6 +1110,11 @@ int mbedtls_aes_crypt_cbc(mbedtls_aes_context *ctx,
     }
 #endif
 
+    /* Nothing to do if length is zero. */
+    if (length == 0) {
+        return 0;
+    }
```

Comment:
I don't understand why this can't be above the Padlock check, after the other length check.

Either Padlock behaves the same as non-Padlock, in which case it makes no difference, or Padlock behaves differently to non-Padlock when length == 0, and we have a pre-existing problem :)

And "nothing to do" shouldn't depend on whether it's accelerated or not

Reply:
I am unsure of what exactly the padlock implementation is doing, but admittedly, it should not be doing anything.